src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -23,6 +24,12 @@ class ChannelCache<T> {
           String, ConcurrentHashMap<String, SnowflakeStreamingIngestChannelInternal<T>>>
       cache = new ConcurrentHashMap<>();
 
+  // Last flush time for each table, the key is FullyQualifiedTableName.
```

Comment:
Lets introduce a TableCache to track per-table information. Most of the changes in this class will end up moving over.

Reply:
nvm, read through the whole thing and what you have looks okay.